include/boost/math/fft/abstract_ring.hpp

```diff
@@ -50,6 +51,27 @@
   */
 
   namespace boost { namespace math {  namespace fft {  namespace detail {
+  
+    #if __cplusplus < 201700L
+    template<typename ...> using void_t = void;
+    #else
+    using std::void_t;
+    #endif
+    
+
+    template<class T, typename = void_t<> >
+    struct is_complex : std::false_type
+    {};
+
+    template<class T >
+    struct is_complex<T, 
+        void_t<
+            typename T::value_type,
+            decltype(sin(std::declval<typename T::value_type>())),
+            decltype(cos(std::declval<typename T::value_type>()))
```

Comment:
Hmm.. if I get the intentions correctly it should be checking for presence of

```
decltype(sin(std::declval<T>())),
decltype(cos(std::declval<T>()))
```

Or did I misunderstood something?

Reply:
I see, you are computing `sin(…)` and `cos(…)` of phase ∈ ℝ. OK. I will need to ponder this for a while.

Reply:
Yes, it is not so important to compute the sin/cos of the complex number. Our implementation need the sin/cos of the underlying real phase.